initial commit приложения Filmorate

Sweta304 opened the pull request.

Sweta304 committed:
initial commit приложения Filmorate

alexache commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,65 @@
+
+    @PutMapping
+    public Film updateFilm(@RequestBody @Valid Film film) throws FilmNotFoundException, ValidationException {
+        if (!films.contains(film)) {
```
(можно лучше) В List<Film> films есть определенные недостатки - для поиска наличия фильма приходится обходить весь список, так же и для обновления фильма. Будет лучше хранить в виде маппинга - id -> объект (то есть в хэш мапе), как в прошлых спринтах. Тогда поиск и обновление будет работать заметно быстрее

Sweta304 replied:
исправлено

alexache commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,65 @@
+        return ++lastId;
+    }
+
+    public List<Film> getFilms() {
```
Действительно ли нужен отдельный метод, почему нельзя воспользоваться в тесте getAllFilms() ?

Sweta304 replied:
убрала, действительно лишний

alexache commented on src/test/java/ru/yandex/practicum/filmorate/FilmorateApplicationTests.java:
```diff
@@ -1,13 +1,274 @@
+import static org.junit.jupiter.api.Assertions.*;
 
 @SpringBootTest
 class FilmorateApplicationTests {
```
(можно лучше) я бы предложил разделить User / Film в разные тест классы, у него ведь нет никакого общего функционала, по крайне мере сейчас

Sweta304 replied:
готово

alexache commented on src/test/java/ru/yandex/practicum/filmorate/FilmorateApplicationTests.java:
```diff
@@ -1,13 +1,274 @@
+        } catch (UserAlreadyExistsException | ValidationException e) {
+            System.out.println(e.getMessage());
+        }
+        assertTrue(userController.getUsers().isEmpty(),"добавлен пользователь с некорректным email");
```
А почему сделано через assertTrue, а не assertThrows(), как выше? Разве заранее не известно, какой из эксепшенов тут возможен?

Sweta304 replied:
Сделала так, чтобы проверить работу аннотации @Email, но поняла сейчас что это вообще лишнее, что так не проверить

alexache commented on src/test/java/ru/yandex/practicum/filmorate/FilmorateApplicationTests.java:
```diff
@@ -1,13 +1,274 @@
+        assertTrue(userController.getUsers().isEmpty(),"добавлен пользователь с некорректным email");
+    }
+
+//    @Test
```
Закомментированный код лучше удалять

Sweta304 replied:
убрала

alexache commented on src/test/java/ru/yandex/practicum/filmorate/FilmorateApplicationTests.java:
```diff
@@ -1,13 +1,274 @@
+    }
+
+    @Test
+    void IncorrectFilmReleaseDate() {
```
Имена методов не стоит начинать с заглавной буквы

Sweta304 replied:
исправлено

Sweta304 replied:
исправлено

alexache commented on src/test/java/ru/yandex/practicum/filmorate/FilmorateApplicationTests.java:
```diff
@@ -1,13 +1,274 @@
+        assertEquals("данные о фильме указаны некорректно", exception.getMessage());
+    }
+
+//    @Test
```
Закомментированный код лучше удалять

Sweta304 replied:
удалила

alexache commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,52 @@
+@Data
+@Builder
+@Validated
+@ConfigurationProperties(prefix="application.properties")
```
А для чего здесь эта аннотация?

Sweta304 replied:
удалила, пробовала экспериментировать, т.к. не работала валидация e-mail

alexache reviewed (changes requested):
Оставил комментарии по коду

Sweta304 committed:
refactor:
- созданы раздельные тестовые классы для каждого контроллера
- удалены неиспользуемые тесты
- хранение пользователей и фильмов теперь в Map вместо List

alexache commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -9,38 +9,41 @@
     @PostMapping
     public Film addFilm(@RequestBody @Valid Film film) throws FilmAlreadyExistsException, ValidationException {
-        if (films.contains(film)) {
+        if (films.containsValue(film)) {
```
(можно лучше) все равно это выглядит немного странным, проход по всей мапе тоже будет накладной операцией, в ТЗ не было про проверку уникальности фильма, поэтому пока можно было пренебречь наверное

alexache reviewed (approved):
Теперь все выглядит ок

Sweta304 committed:
refactor:
- изменена проверка на наличие фильма/пользователя
